content/doc/developer/security/scan.adoc

```diff
@@ -0,0 +1,53 @@
+---
+title: Jenkins Security Scan
+layout: developerguide
+references:
+- title: Introductory blog post (2020)
+  url: https://www.jenkins.io/blog/2020/11/04/codeql/
+- title: GitHub CodeQL Repository
+  url: https://github.com/github/codeql
+- title: Jenkins CodeQL Repository
+  url: https://github.com/jenkins-infra/jenkins-codeql
+- title: Jenkins Security Scan Repository
+  url: https://github.com/jenkins-infra/jenkins-security-scan
+---
+
+== Motivation
+
+Jenkins uses the https://github.com/jenkinsci/stapler[Stapler web framework] to process HTTP requests, instead of a more common framework.
+As a result, generic static analysis tools are often unable to find common vulnerabilities in Jenkins core and plugins.
+
+To address this problem, the link:/security/team/[Jenkins security team] created a custom code scanner, built with GitHub's https://github.com/github/codeql[CodeQL], to find vulnerabilities commonly occurring in Jenkins plugins.
+
+This page explains how to set up code scanning with this tool.
+
+== Usage in a Jenkins project-hosted plugin
+
+In your GitHub repository, select the "Actions" link on top.
+In the section "By Jenkins", select "Jenkins Security Scan" and configure the YAML file in your repository.
+Once saved, you should be able to see findings in the "Security" tab under "Code scanning alerts", and pull requests will have an additional check.
+
+////
+TODO Decide whether we want to continue to support this.
+== Usage in a Jenkins project-hosted plugin (GitHub topic)
+
+Add the topic `jenkins-security-scan-enabled` to your repository.
+Within 24 hours, the first code scanning results should appear.
+Unlike the GitHub Action, feedback is not immediate, as scans are only running periodically.
+////
+
+== Usage outside the Jenkins project
+
+You can use the same GitHub workflow and action in your Jenkins plugin project hosted on GitHub even if it's outside the `jenkinsci` organization.
+Just copy the YAML workflow template from https://github.com/jenkinsci/.github/tree/master/workflow-templates[here] into your repository's `.github/workflows` directory, replacing the `$default-branch` placeholder as appropriate.
```

Comment:
https://webaccess.berkeley.edu/ask-pecan/click-here
```suggestion
Just copy the YAML workflow template from https://github.com/jenkinsci/.github/tree/master/workflow-templates[jenkinsci/.github] into your repository's `.github/workflows` directory, replacing the `$default-branch` placeholder as appropriate.
```

Reply:
Good point, I generally try to not do this 😅